source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -275,9 +296,11 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
     const std::string escaped_redirect_uri =
         Http::Utility::PercentEncoding::encode(redirect_uri, ":/=&?");
 
+    const std::string escaped_auth_scopes = absl::StrJoin(config_->authScopes(), "%20");
```

Comment:
Should the scopes also be escaped (i.e., joined with spaces and using `Http::Utility::PercentEncoding::encode` to escape the string)?

Reply:
@adisuissa I tried to join the list with whitespace and use PercentEncoding::encode, but this function seems to do not replace the whitespace character with percent. Then I just used %20 instead. Do you mean something different?

Reply:
Can the names of the auth scopes require escaping as well? You should be able to include ` ` as a character in the second arg to `encode` to have it escape it